lib/cn/kvs_mblk_desc.c

```diff
@@ -0,0 +1,104 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (C) 2022 Micron Technology, Inc.  All rights reserved.
+ */
+
+#define MTF_MOCK_IMPL_mblk_desc
+
+#include <sys/mman.h>
+
+#include <mpool/mpool_structs.h>
+#include <mpool/mpool.h>
+
+#include <hse/error/merr.h>
+
+#include <hse_util/assert.h>
+#include <hse_util/compiler.h>
+#include <hse_util/event_counter.h>
+#include <hse_util/page.h>
+
+#include "kvs_mblk_desc.h"
+
+merr_t
+mblk_mmap(struct mpool *mp, uint64_t mbid, struct kvs_mblk_desc *md)
+{
+    merr_t err;
+    struct mblock_props props;
+    const void *base;
+
+    err = mpool_mblock_props_get(mp, mbid, &props);
+    if (ev(err))
+        return err;
+
+    err = mpool_mblock_mmap(mp, mbid, &base);
+    if (ev(err))
+        return err;
+
+    assert(props.mpr_alloc_cap % PAGE_SIZE == 0);
+    assert(props.mpr_write_len % PAGE_SIZE == 0);
+
+    md->map_base = (void *)base;
+    md->alen_pages = props.mpr_alloc_cap / PAGE_SIZE;
+    md->wlen_pages = props.mpr_write_len / PAGE_SIZE;
+    md->mclass = props.mpr_mclass;
+    md->mbid = mbid;
+
+    /* Verify mappings to pages and smaller int types don't lose information.
+     */
+    assert(md->alen_pages * PAGE_SIZE == props.mpr_alloc_cap);
+    assert(md->wlen_pages * PAGE_SIZE == props.mpr_write_len);
+    assert(md->mclass == props.mpr_mclass);
```

Comment:
You could static_assert this last one

static_assert(UINT8_MAX > MCLASS_MAX) or whatever